src/data/rel.lean

```diff
@@ -9,27 +9,32 @@ import tactic.basic data.set.lattice order.complete_lattice logic.relator
 
 variables {α : Type*} {β : Type*} {γ : Type*}
 
+/-- A relation on `α` and `β`, aka a set-valued function, aka a partial multifunction --/
 @[derive lattice.complete_lattice]
 def rel (α : Type*) (β : Type*) := α → β → Prop
 
 namespace rel
 
 variables {δ : Type*} (r : rel α β)
 
+/-- The inverse relation : `r.inv x y ↔ r y x`. Note that this is *not* a groupoid inverse. -/
 def inv : rel β α := flip r
 
 lemma inv_def (x : α) (y : β) : r.inv y x ↔ r x y := iff.rfl
 
 lemma inv_inv : inv (inv r) = r := by { ext x y, reflexivity }
 
+/-- Domain of a relation -/
 def dom := {x | ∃ y, r x y}
 
+/-- Codomain aka range of a relation-/
 def codom := {y | ∃ x, r x y}
 
 lemma codom_inv : r.inv.codom = r.dom := by { ext x y, reflexivity }
 
 lemma dom_inv : r.inv.dom = r.codom := by { ext x y, reflexivity}
 
+/-- Composition of relation; note that it follows the category theory order of arguments. -/
```

Comment:
```suggestion
/-- Composition of relation; note that it follows the `category_theory/` order of arguments. -/
```
I suggest formatting this as a library name, to make clear that this is note exactly what 99% of category theorists do. (Just to be clear :wink:: I'm not making any judgment here!)